source-stripe-native/source_stripe_native/models.py

```diff
@@ -151,6 +151,8 @@ class Accounts(BaseStripeObjectWithEvents):
         "account.updated": "u",
     }
 
+    # Accounts docs returned in account.updated events may not have a created field.
+    created: int | None = None
```

Comment:
Is there a way to get this to schematize as "not required to be present, but if it is present, it must be `int`"? I think that's what we want: The field is either present as an int, or not present at all. It shouldn't ever be present as an actual `null` value.

The schema coming from this is saying it _can_ be a `null`, and also giving it a default `null` value (we don't want that default, for sure). Ideally the schema would be this:

```
created: {
  "type": "integer",
  "title": "created"
}
```

...and it would not be present in the list of `required` fields.

Reply:
I've looked through a handful of Pydantic discussions, and I haven't found any straightforward way to have an optional, non-nullable field. I've looked in Pydantic's [docs](https://docs.pydantic.dev/latest/migration/#required-optional-and-nullable-fields), and it looks like optional fields always have to have a default value.

To avoid saying `created` can be `null` in the schema, we could have a default value like `-1`, then overwrite `created` with the event's timestamp whenever we see that it's `-1`. That's not 100% ideal, but I'm not seeing a better way to handle this right now. What do you think about that approach?

Reply:
I have used [this hack](https://github.com/estuary/connectors/blob/911128ff1037dbaecc7bc18082fb085c13717291/source-hubspot-native/source_hubspot_native/models.py#L400) for not schematizing the default value before - does that help here?

I think we need a way to represent this correctly. If it the field is either present and is some integer, or is not present at all, we need to have the schema say that. A default of `null` is not ideal, but a default of `-1` is even worse since that is not what the value of the field is, and since it's a unix timestamp might be problematic in other ways.

Reply:
That code you linked did help - thank you! I'll get the PR updated.